tests/v2/validation/KDM/kdm_test.go

```diff
@@ -0,0 +1,228 @@
+//go:build validation
+
+package KDM
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	stevev1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	rancherDeployment    = "rancher"
+	rancherNamespace     = "cattle-system"
+	rancherLabelSelector = "app=rancher"
+	rkeMetadataConfig    = "rke-metadata-config"
+)
+
+var defaultBackoff = wait.Backoff{
+	Duration: 1 * time.Second,
+	Factor:   1.5,
+	Steps:    10,
+}
+
+type KDMTestSuite struct {
+	suite.Suite
+	client    *rancher.Client
+	session   *session.Session
+	cluster   *management.Cluster
+	config    *rest.Config
+	clientset *kubernetes.Clientset
+}
+
+func (k *KDMTestSuite) SetupSuite() {
+	var err error
+	k.session = session.NewSession()
+	k.client, err = rancher.NewClient("", k.session)
+	k.Require().NoError(err, "Failed to create RancherClient")
+	k.cluster, err = k.client.Management.Cluster.ByID("local")
+	k.Require().NoError(err)
+	k.Require().NotEmpty(k.cluster)
+	localClusterKubeconfig, err := k.client.Management.Cluster.ActionGenerateKubeconfig(k.cluster)
+	k.Require().NoError(err)
+	c, err := clientcmd.NewClientConfigFromBytes([]byte(localClusterKubeconfig.Config))
+	k.Require().NoError(err)
+	k.config, err = c.ClientConfig()
+	k.Require().NoError(err)
+	k.clientset, err = kubernetes.NewForConfig(k.config)
+	k.Require().NoError(err)
+}
```

Comment:
For this SetupSuite, after the `require()`'s following each call, could i request you add a new line to better format the spacing w/ this, so at a glance its easier to follow 🙂